src/llm/continuous_batching/servable.cpp

```diff
@@ -0,0 +1,131 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include "../../logging.hpp"
+#include "../../status.hpp"
+
+#pragma warning(push)
+#pragma warning(disable : 4005 4309 6001 6385 6386 6326 6011 4005)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_graph.h"
+#pragma GCC diagnostic pop
+#pragma warning(pop)
+
+#include "../../http_payload.hpp"
+#include "../../mediapipe_internal/mediapipe_utils.hpp"
+#include "../apis/openai_completions.hpp"
+#include "../text_processor.hpp"
+#include "llm_executor.hpp"
+#include "servable.hpp"
+
+namespace ovms {
+
+void ContinuousBatchingServable::notifyExecutorThread() {
+    auto properties = std::static_pointer_cast<ContinuousBatchingServableProperties>(this->properties);
+    SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Notifying executor thread");
+    if (properties->llmExecutorWrapper == nullptr) {
+        SPDLOG_LOGGER_ERROR(llm_calculator_logger, "LLMExecutorWrapper is not initialized");
+        return;
+    }
+    properties->llmExecutorWrapper->notifyNewRequestArrived();
+}
+
+// Node resources interface start
+
+ovms::Status ContinuousBatchingServable::initialize() {
+    auto properties = std::static_pointer_cast<ContinuousBatchingServableProperties>(this->properties);
+    if (properties->pipeline == nullptr) {
+        throw std::logic_error("Cannot initiate generation with uninitialized pipeline");
+    }
+    properties->llmExecutorWrapper = std::make_shared<LLMExecutorWrapper>(properties->pipeline);
```

Comment:
we have special initializers now - why not initialize it there?

Reply:
Done